src/main/java/com/oracle/nosql/spring/data/core/NosqlTemplateBase.java

```diff
@@ -178,6 +188,181 @@ protected boolean doCreateTableIfNotExists(
+                msg = String.format("Shard primary keys mismatch: " +
+                        "table=%s, entity=%s.", tableShards, entityShards);
+                errors.add(msg);
+            }
+
+            // check non-shard keys and types match
+            if (!tableNonShards.equals(entityNonShards)) {
+                msg = String.format("Non-shard primary keys mismatch: " +
+                                "table=%s, entity=%s.", tableNonShards,
+                        entityNonShards);
+                errors.add(msg);
+            }
+
+            // check non-primary keys and types match
+            if (!tableOthers.equals(entityOthers)) {
+                msg = String.format("Non-primary key columns mismatch:" +
+                                "table=%s, entity=%s.", tableOthers,
+                        entityOthers);
+                errors.add(msg);
+            }
+
+            // check identity same
+            FieldValue identity = tableSchema.get(identityField);
+            if (identity != null && !entityInformation.isAutoGeneratedId()) {
+                errors.add("Identity information mismatch.");
+
+            } else if (identity == null && entityInformation.isAutoGeneratedId() &&
+                    entityInformation.getIdNosqlType() != FieldValue.Type.STRING) {
+                errors.add("Identity information mismatch.");
+            }
+
+            // TTL warning
+            FieldValue ttlValue = tableSchema.get(ttlField);
+            TimeToLive ttl = entityInformation.getTtl();
+            // TTL is present in database but not in the entity
+            if (ttlValue != null && ttl != null &&
+                    !ttl.toString().equalsIgnoreCase(ttlValue.getString())) {
+                LOG.warn("TTL of the table in database is different from " +
+                        "the TTL of the entity " +
+                        entityInformation.getJavaType().getName());
+            } else if (ttlValue == null && ttl != null && ttl.getValue() != 0) {
+                // TTL is present in entity but not in the database
+                LOG.warn("TTL of the table in database is different from " +
+                        "the TTL of the entity " +
+                        entityInformation.getJavaType().getName());
+            }
+        } catch (NullPointerException | ClassCastException ex) {
+            // something is wrong in parsing json schema
+            String msg = String.format("Could not validate schema of the " +
+                            "table %s in the database against entity %s : " +
+                            "%s",
+                    entityInformation.getTableName(),
+                    entityInformation.getJavaType().getName(),
+                    ex.getMessage());
+
+            if (LOG.isDebugEnabled()) {
+                LOG.debug("JSON Schema of the table is " + jsonSchema);
+            }
+            throw new IllegalStateException(msg, ex);
```

Comment:
Maybe log warn is a better choice here, in case the format changes but the table stays the same.

Reply:
Done. Changed exception to warning log.